### PR TITLE
Update blog details footer section with optional extra spacing between sections

### PR DESCRIPTION
**Pull request by jaclync**
Refs. #11467 - subtask - [design polish] Add extra 20pt spacing after footer if the next section has title

* Created `BlogDetailsSectionFooterView` that contains a title label and a spacer view (20px height) in a vertical stack view. Configuration method updates the title text, and whether the spacer view is shown.
* In `BlogDetailsViewController`, replaced default footer title setup with `BlogDetailsSectionFooterView`, and set to show extra spacing when the next section has no-empty title.

To test:
- Turn on Domain Credit UI by returning true in `DomainCreditEligibilityChecker.canRedeemDomainCredit`
- Go to a site
- Turn `QuickStart` on (if not already on, can return true in `BlogDetailsViewController`'s `shouldShowQuickStartChecklist`)
  - There should be comfortable spacing (more than 20px) between the end of Domain Credit footer title and the the title of QuickStart ("NEXT STEPS")
- Turn `QuickStart` off (if not already off, can return false in `BlogDetailsViewController`'s `shouldShowQuickStartChecklist`)
  - There should be similar spacing between the end of Domain Credit footer title and the the beginning of the next section

Example screenshots:

next section without title | next section with title
--- | ---
Simulator Screen Shot - iPhone SE - 2019-05-01 at 23 05 50 | Simulator Screen Shot - iPhone SE - 2019-05-01 at 23 08 54

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/BlogDetailsSectionFooterView.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+class BlogDetailsSectionFooterView: UITableViewHeaderFooterView {
+    private let titleLabel: UILabel = {
+        let titleLabel = UILabel()
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        titleLabel.numberOfLines = 0
+        titleLabel.font = WPStyleGuide.tableviewSectionFooterFont()
+        titleLabel.textColor = WPStyleGuide.greyDarken10()
+        return titleLabel
+    }()
+    private let spacerView: UIView = UIView(frame: .zero)
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Blog/BlogDetailsSectionFooterView.swift]: You can omit the `: UIView` and let the compiler infer the type. I used to define the type every time but compact code is even better 😊
```suggestion
    private let spacerView = UIView(frame: .zero)
```

File: WordPress/Classes/ViewRelated/Blog/BlogDetailsSectionFooterView.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupSubviews()
+    }
+
+    private func setupSubviews() {
+        let stackView = UIStackView(arrangedSubviews: [titleLabel, spacerView])
+        stackView.translatesAutoresizingMaskIntoConstraints = false
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Blog/BlogDetailsSectionFooterView.swift]: `UIStackView` is a good solution here 👍

File: WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m [hunk 3]
```diff
@@ -540,6 +542,15 @@ - (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSIntege
 }
 
 - (UIView *)tableView:(UITableView *)tableView viewForFooterInSection:(NSInteger)section {
+    BlogDetailsSection *detailSection = self.tableSections[section];
+    NSString *footerTitle = detailSection.footerTitle;
+    if (footerTitle != nil) {
+        BlogDetailsSectionFooterView *footerView = (BlogDetailsSectionFooterView *)[tableView dequeueReusableHeaderFooterViewWithIdentifier:BlogDetailsSectionFooterIdentifier];
+        // If the next section has title, gives extra spacing between two sections.
+        BOOL doesNextSectionHaveTitle = (self.tableSections.count > section + 1) ? (self.tableSections[section + 1].title != nil): NO;
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m]: What if we call this `BOOL` like the method argument? I like what you used in the row below: `shouldShowExtraSpacing`

**Comment by SylvesterWilmott** [comment]
Looks good @jaclync ! Thanks

**Pull request merged**